app/src/main/java/org/openobservatory/ooniprobe/common/Application.java

```diff
@@ -35,26 +38,35 @@ public class Application extends android.app.Application {
 	@Inject Gson _gson;
 	@Inject OkHttpClient _okHttpClient;
 	@Inject OONIAPIClient _apiClient;
+	ExecutorService executorService = Executors.newFixedThreadPool(4);
 
 	public AppComponent component;
 
 	@Override public void onCreate() {
 		super.onCreate();
+		TimingLogger timings = new TimingLogger(Application.class.getSimpleName(), "app-on-create");
 		component = buildDagger();
 		component.inject(this);
+		timings.addSplit("dagger-init");
 
 		FlowManager.init(this);
 		if (BuildConfig.DEBUG)
 			FlowLog.setMinimumLoggingLevel(FlowLog.Level.V);
+		timings.addSplit("flow-init");
 		AppLifecycleObserver appLifecycleObserver = new AppLifecycleObserver();
 		ProcessLifecycleOwner.get().getLifecycle().addObserver(appLifecycleObserver);
-		if (_preferenceManager.canCallDeleteJson())
-			Measurement.deleteUploadedJsons(this);
-		Measurement.deleteOldLogs(this);
-		ThirdPartyServices.reloadConsents(this);
-
+		timings.addSplit("lifecycle-init");
+		executorService.execute(() -> {
+			if (_preferenceManager.canCallDeleteJson())
+				Measurement.deleteUploadedJsons(Application.this);
+			Measurement.deleteOldLogs(Application.this);
+			ThirdPartyServices.reloadConsents(Application.this);
```

Comment:
```Java
ThirdPartyServices.reloadConsents(Application.this);
```

I am not convinced by this line. For two reasons:

1. are we sure that it's safe to initialize third party services from a background thread while the previous semantics was that the rest of the app _happens after_ we have initialized third party services?
2. what if the current services support this kind of async initialization and everything is WAI _but_ in the future we add a new third-party service behind the opaque `reloadConsents` API and we break it? What ways we have to ensure our future selves would not commit this mistake? Would documentation be enough for this?

Additionally, I think that, conceptually, third-party services are lifecycle-init and not measurement-init.